lua/refactoring/point.lua

```diff
@@ -64,8 +58,41 @@ function Point:clone()
     return clone
 end
 
-function Point:within_node(node)
-    return Region:from_node(node, 0):contains_point(self) -- buffer doesn't matter in this case
+--- Compare the position of two points.
+-- Given two points with positions pos_1 and pos_2
+-- return -1 if pos_1 < pos_2, 0 if pos_1 == pos_2 and 1 otherwise.
+-- @params point   the second point to compare to.
+-- @return int     either -1, 0 or 1
+function Point:compareTo(point)
```

Comment:
could you project this to 1d

```
local function project_point(point)
    return point.row * 10000000 + point.col
end
```

That way we don't have to have a complicated `compareTo` function

Then also rename `compareTo` to `compare_to` for lua conventions.

Also we need some tests because `0 != false` in lua.  Therefore you have some functions that are incorrect.  `not 0 == false`.  Which means you have to do `... == 0` comparisons.

Reply:
Good point about the renaming! Will take care of that, I also just realised there is a simpler way of writing the function, will push that soon so that you can take a look. I personally think is much easier to think about simple if-statement comparisons but it seems you are convinced of the 1d-projection idea. Is there a reason for this? Would it be a bad idea to just keep the if-statement comparisons?

Reply:
I don't think I am making any 0 == false comparisons in the code but it's a good point and I certainly didn't know that 0 is not falsy in lua. I will definitely write some tests and update the PR.

Reply:
Tests great for all the new helper functions .  I think it will make sure that we don't have any oopsies and we already have some tests so it should be super simple to create them.

On projection point of view.  The reason why its nicer is that it turns a 5 branch function into a 3 branch.  Its just simpler code to look at.  at this point I am not strong enough against it, so whatever.

Reply:
Thank you for your flexibility, Prime! I have added the tests and fixed the merge conflicts, I think this should be okay now :) let me know what you think